source/simulator/simulator_access.cc

```diff
@@ -579,14 +579,36 @@ namespace aspect
   }
 
 
+
+  template <int dim>
+  std::shared_ptr<const InitialTemperature::Manager<dim>>
+  SimulatorAccess<dim>::get_initial_temperature_manager_pointer () const
+  {
+    Assert (simulator->initial_temperature_manager,
+            ExcMessage ("You are trying to access the initial temperature manager "
+                        "object, but the Simulator object is no longer keeping "
+                        "track of it because the initial time has passed. If "
+                        "you need to access to this object after the first time "
+                        "step, you need to copy the object returned by "
+                        "this function before or during the first time step "
+                        "into a std::shared_ptr that lives long enough to "
+                        "extend the lifetime of the object pointed to "
+                        "beyond the timeframe that the Simulator object "
+                        "keeps track of it."));
+    return simulator->initial_temperature_manager;
+  }
+
+
+
   template <int dim>
   const InitialTemperature::Manager<dim> &
   SimulatorAccess<dim>::get_initial_temperature_manager () const
   {
-    return simulator->initial_temperature_manager;
+    return *get_initial_temperature_manager_pointer();
```

Comment:
This will always copy the shared pointer, modify the ref count, etc..
Wouldn't it make sense to access it directly from the simulator here?

Reply:
Yes. I didn't see another way: Right now we have
```
      std::shared_ptr<InitialTemperature::Manager<dim>>                      initial_temperature_manager;
```
and
```
      std::shared_ptr<const InitialTemperature::Manager<dim>>
      get_initial_temperature_manager_pointer () const;
```
The pointed to object is non-`const` in `class Simulator`, but we want to return a pointer to `const`. The current declaration of `get_initial_temperature_manager_pointer()` does that. The only other option I see is to let the function return a `const` reference to the non-`const` object.

...

...

Well, of course I could also just let the function that returns the reference to the object do the access itself and not go through the `_pointer()` function. On second thought, that's exactly what you are proposing -- I guess I just needed to think that through myself :-)

So implemented now!